air-quality-backend/src/air_quality/etl/forecast/forecast_adapter.py

```diff
@@ -107,3 +113,309 @@ def transform(forecast_data: ForecastData, locations: list[AirQualityLocation])
+
+    :param rgb_data_array: array with shape (num_lat, num_lon * num_time, channels)
+    :param num_time_steps: Number of time steps to use for each chunk
+    :param num_lon: Number of longitude points
+    :param time_vector: DataArray containing time information for each time step
+    :return: A list of chunked arrays and a dictionary with start and end time
+    stamps for each chunk
+    """
+    num_lat, total_lon_time, channels = rgb_data_array.shape
+    num_total_time_steps = total_lon_time // num_lon
+
+    chunks = []
+    time_steps_dict = {}
+
+    if num_time_steps <= 0 or num_time_steps > num_total_time_steps:
+        num_time_steps = num_total_time_steps
+
+    for i in range(0, num_total_time_steps, num_time_steps):
+        start_time_step = i
+        end_time_step = min(i + num_time_steps, num_total_time_steps)
+
+        start_index = start_time_step * num_lon
+        end_index = end_time_step * num_lon
+
+        chunk = rgb_data_array[:, start_index:end_index, :]
+        chunks.append(chunk)
+
+        time_start = pd.to_datetime(
+            time_vector.data[start_time_step:end_time_step].min(), unit="s"
+        ).strftime("%Y-%m-%d %H:%M:%S")
+        time_end = pd.to_datetime(
+            time_vector.data[start_time_step:end_time_step].max(), unit="s"
+        ).strftime("%Y-%m-%d %H:%M:%S")
+        time_steps_dict[len(chunks) - 1] = {
+            "time_start": time_start,
+            "time_end": time_end,
+        }
+
+    return chunks, time_steps_dict
+
+
+def _create_output_directory(forecast_data: ForecastData) -> Tuple[str, str]:
+    """
+    Create output directory for data textures.
+
+    :param forecast_data: The forecast data object.
+
+    :return: The output directory path and forecast date.
+    """
+
+    forecast_date = datetime.fromtimestamp(
+        forecast_data.get_time_value(), timezone.utc
+    ).strftime("%Y-%m-%d_%H")
+
+    output_directory = f"/app/data_textures/{forecast_date}"
+    if not os.path.exists(output_directory):
+        output_directory = f"{os.getcwd()}/data_textures/{forecast_date}"
+    os.makedirs(output_directory, exist_ok=True)
+    return output_directory, forecast_date
+
+
+def _save_data_texture(
+    rgb_data_array: np.ndarray,
+    output_directory: str,
+    forecast_date: str,
+    variable: str,
+    num_chunk: str,
+    total_chunks: str,
+    file_format: str,
+) -> str:
+    """
+    Save data textures to disk and convert to binary data.
+
+    :param rgb_data_array: Numpy array containing the image data.
+    :param output_directory: Directory to save the textures.
+    :param forecast_date: Forecast date string.
+    :param variable: Variable name to include in the filename.
+    :param num_chunk: Chunk number.
+    :param total_chunks: Total number of chunks.
+    :param file_format: File format for saving the texture.
+
+    :return: The path to the saved file.
+    """
+    if variable == "winds_10m":
```

Comment:
We should move some of these hardcoded strings into a const variable, winds_10m for example is repeated a fair amount I think

Reply:
added some constants at the top of the script